src/components/Notifications/EventLog/usePrimaryToolbarFilterConfigWrapper.tsx

```diff
@@ -0,0 +1,180 @@
+import { ColumnsMetada, usePrimaryToolbarFilterConfig } from '@redhat-cloud-services/insights-common-typescript';
+import produce from 'immer';
+import React from 'react';
+import { useMemo } from 'react';
+
+import { Schemas } from '../../../generated/OpenapiNotifications';
+import { ClearEventLogFilters, EventLogFilterColumn, EventLogFilters, SetEventLogFilters } from './EventLogFilter';
+import { EventLogTreeFilter } from './EventLogTreeFilter';
+
+export interface EventLogCustomFilter {
+    bundleId: string,
+    category: string,
+    chips: Array<{ name: string, value: string, isRead: boolean }>
+}
+
+// Wrapper hook that gets the PrimaryToolbarFilterConfig and adds a custom conditional filter using Dropdown/Tree components
+// usePrimaryToolbarFilterConfig only supports 3 filter types: checkbox, radio, and text, so this extends that
+export const usePrimaryToolbarFilterConfigWrapper = (
+    bundles: readonly Schemas.Facet[],
+    filters: EventLogFilters,
+    setFilters: SetEventLogFilters,
+    clearFilter: ClearEventLogFilters,
+    metaData: ColumnsMetada<typeof EventLogFilterColumn>
+) => {
+    const [ customFilters, setCustomFilters ] = React.useState([] as EventLogCustomFilter[]);
+    const toolbarConfig = usePrimaryToolbarFilterConfig(
+        EventLogFilterColumn,
+        filters,
+        setFilters,
+        clearFilter,
+        metaData
+    );
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const defaultDelete = React.useMemo(() => toolbarConfig.activeFiltersConfig.onDelete, []);
+    toolbarConfig.activeFiltersConfig.onDelete = React.useCallback((
+        _event: any,
+        rawFilterConfigs: any[]
+    ) => {
+        try {
+            defaultDelete(_event, rawFilterConfigs);
```

Comment:
Instead of this approach, could we try: 
. Detect if the `rawFilterConfigs` has any detail that is important for us.
  . If has any, process and strip that from the `rawFilterConfigs` (create a new object)
. process the `defaultDelete`

My worry is that the catch would swallow other errors that would go unnoticed

Reply:
That's a good catch, however, I've looked through the original onDelete function and there are no errors that are thrown outside of `unexpected column/filter label` which are derived on the metadata. Since the custom filter is not in the metadata it has no purpose calling that function.

But this point does bring up the case where the existing 'non custom' filters (that do exist in the metadata) cause an error... the current implementation will assume that it was a custom filter causing the error, which is wrong. I will change to accommodate that, but I don't see a reason for the custom filter deletes to call the original one.

Reply:
Actually, now I understand what you are referring to, if you were to clear filters with a mix of custom and non custom, you would need to go through both delete functionality

Reply:
Implemented more robust solution following the pseudo code in the parent comment